### DISCUSSION TITLE
meson test failed about su2@master on canetos8_aarch64

### DISCUSSION THREAD

**Discussion by Tom-python0121** [su2code/SU2#1867] [Q&A]
Hello, I meet a problem:meson test failed about su2@master on canetos8_aarch64.
```console
[root@bigdata spack-build]# ls
Common  SU2_CFD  SU2_DEF  SU2_DOT  SU2_GEO  SU2_PY  SU2_SOL  UnitTests  build.ninja  compile_commands.json  externals  meson-info  meson-logs  meson-private
[root@bigdata spack-build]# meson test
ninja: Entering directory `/home/stage/root/spack-stage-su2-master-6pqdoqje4yuyseje7lkxnucdij7l5njz/spack-src/spack-build'
ninja: no work to do.
1/1 Catch2 test driver        FAIL            0.01s   exit status 1
>>> MALLOC_PERTURB_=206 /home/stage/root/spack-stage-su2-master-6pqdoqje4yuyseje7lkxnucdij7l5njz/spack-src/spack-build/UnitTests/test_driver



Ok:                 0
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0

Full log written to /home/stage/root/spack-stage-su2-master-6pqdoqje4yuyseje7lkxnucdij7l5njz/spack-src/spack-build/meson-logs/testlog.txt
```
testlog details:
[testlog.txt](https://github.com/su2code/SU2/files/10323737/testlog.txt)

**Comment by pcarruscag** [comment]
Does the code run?
What happens if you use ./meson.py instead of just meson?
What happens if you call the unit test executables directly (they will be in the meson build folder)?

**Comment by Tom-python0121** [comment]
@pcarruscag This is compiled using the spack management and has been run. The previous version can run successfully, but it fails after version 7.5.0

**Comment by pcarruscag** [comment]
previously when?

**Comment by Tom-python0121** [comment]
@pcarruscag   su2@7.3.1 can passed.
![image](https://user-images.githubusercontent.com/77763324/210057537-a61645f2-adc1-431a-be21-8a18015ab8d2.png)

**Comment by pcarruscag** [comment]
I'm not sure that is running the SU2 unit tests.
Can you run the test_driver executable directly and check the output?
It will be in the "build" directory / UnitTests/test_driver

**Comment by Tom-python0121** [comment]
@pcarruscag There is no indication that there is a problem with the test_driver execution.
[root@bigdata spack-build]# cd UnitTests/
[root@bigdata UnitTests]# ls
test_driver  test_driver.p
[root@bigdata UnitTests]# ./test_driver
[root@bigdata UnitTests]#

**Comment by pcarruscag** [comment]
Thanks for checking, in that case I do not think it is a bug on our side.

**Comment by simon28li** [comment]
Hi, I also repeated this problem when I was testing version 8.0.0 recently. After troubleshooting, two of the tests failed. Can you help me to see what the problem is? the two files involved also exist. In addition, it is really not intuitive that the execution of the command does not directly indicate failure or success.
```
[root@localhost UnitTests]#./test_driver "LUTreader_3D"
Filters: LUTreader_3D
Loading lookup table, filename = src/SU2/UnitTests/Common/containers/lookuptable_3D.drg ...


Error in "void CFileReaderLUT::ReadRawLUT(const string&)":
-------------------------------------------------------------------------
There is no look-up-table file called src/SU2/UnitTests/Common/containers/lookuptable_3D.drg
------------------------------ Error Exit -------------------------------


[root@localhost UnitTests]#./test_driver "LUTreader"
Filters: LUTreader
Loading lookup table, filename = src/SU2/UnitTests/Common/containers/lookuptable.drg ...


Error in "void CFileReaderLUT::ReadRawLUT(const string&)":
-------------------------------------------------------------------------
There is no look-up-table file called src/SU2/UnitTests/Common/containers/lookuptable.drg
------------------------------ Error Exit -------------------------------


[root@localhost UnitTests]#./test_driver
[root@localhost UnitTests]#echo $?
1
```

**Comment by bigfooted** [comment]
`src/SU2/UnitTests/Common/containers/lookuptable_3D.drg`
This path is used by the tests on github and hardcoded in `UnitTests/Common/containers/CLookupTable_tests.cpp`. I think for you this path will simply be `Common/containers/lookuptable_3D.drg ` because you are already in the UnitTests folder. You could create the folder `src/SU2/UnitTests/Common/containers` in the UnitTests folder and copy the files there to pass the tests.

I guess local users will mostly start from the folder UnitTests, so we could add a check to see if we need to start looking in src/ or in Common/
